src/aspire/volume/volume_synthesis.py

```diff
@@ -0,0 +1,352 @@
+        # its opposite faces, and 6 C2 rotation groups whose axes pass through the midpoints of two of
+        # its opposite edges.
+
+        # C4 rotation vectors, ie angle * axis
+        axes_C4 = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]], dtype=self.dtype)
+        angles_C4 = np.array([np.pi / 2, np.pi, 3 * np.pi / 2], dtype=self.dtype)
+        rot_vecs_C4 = np.array(
+            [angle * axes_C4 for angle in angles_C4], dtype=self.dtype
+        ).reshape((9, 3))
+
+        # C3 rotation vectors
+        axes_C3 = np.array(
+            [[1, 1, 1], [-1, 1, 1], [1, -1, 1], [1, 1, -1]], dtype=self.dtype
+        )
+        axes_C3 /= np.linalg.norm(axes_C3, axis=-1)[..., np.newaxis]
+        angles_C3 = np.array([2 * np.pi / 3, 4 * np.pi / 3], dtype=self.dtype)
+        rot_vecs_C3 = np.array(
+            [angle * axes_C3 for angle in angles_C3], dtype=self.dtype
+        ).reshape((8, 3))
+
+        # C2 rotation vectors
+        axes_C2 = np.array(
+            [[1, 1, 0], [-1, 1, 0], [1, 0, 1], [-1, 0, 1], [0, 1, 1], [0, -1, 1]],
+            dtype=self.dtype,
+        )
+        axes_C2 /= np.linalg.norm(axes_C2, axis=-1)[..., np.newaxis]
+        rot_vecs_C2 = np.pi * axes_C2
+
+        # The full set of rotation vectors inducing octahedral symmetry.
+        rot_vec_I = np.zeros((1, 3), dtype=self.dtype)
+        rot_vecs = np.concatenate(
+            (rot_vec_I, rot_vecs_C4, rot_vecs_C3, rot_vecs_C2), dtype=self.dtype
+        )
+
+        # Generate rotations.
+        rots_O = Rotation.from_rotvec(rot_vecs, dtype=self.dtype).matrices
+
+        # Populate coordinates for Gaussian blobs.
+        Q_rot = np.zeros((24 * self.K, 3, 3)).astype(self.dtype)
+        D_sym = np.zeros((24 * self.K, 3, 3)).astype(self.dtype)
+        mu_rot = np.zeros((24 * self.K, 3)).astype(self.dtype)
+        idx = 0
+        for rot in rots_O:
+            for k in range(self.K):
+                Q_rot[idx] = rot.T @ Q[k]
+                D_sym[idx] = 1 / 24 * D[k]
+                mu_rot[idx] = rot.T @ mu[k]
+                idx += 1
+
+        return Q_rot, D_sym, mu_rot
+
+
+class AsymmetricVolume(CnSymmetricVolume):
+    """
+    An asymmetric Volume constructed of random 3D Gaussian blobs with compact support in the unit sphere.
+    """
+
+    def __init__(self, L, C, K=16, seed=None, dtype=np.float64):
+        super().__init__(L=L, C=C, order=1, seed=seed, dtype=dtype)
+
+    def _check_order(self):
+        if self.order != 1:
+            raise ValueError(
+                f"An {self.__class__.__name__} must have order=1. Provided order was {self.order}"
+            )
+
+    def _symmetrize_gaussians(self, Q, D, mu):
+        return Q, D, mu
+
+
+class LegacyVolume(AsymmetricVolume):
+    """
+    An asymmetric Volume object used for testing of legacy code.
+    """
+
+    def generate(self):
+        """
+        Generates a SyntheticVolume with specified symmetry that is multiplied by a bump function
```

Comment:
I believe this comment is incorrect.

I think here we are _not_ using symmetry or clamping with bump function.